src/hubdata/connect_hub.py

```diff
@@ -78,33 +79,100 @@ def __init__(self, hub_path: str | Path):
+        during dataset file discovery.
+
+        :param: ignore_files a str list of file **names** (not paths) or file **prefixes** to ignore when discovering
+            model output files to include in dataset connections. Parent directory names should not be included. The
+            default is to ignore the common files `"README"` and `".DS_Store"`, but additional files can be excluded by
+            specifying them here.
         :return: a pyarrow.dataset.Dataset for my model_output_dir
         """
         # create the dataset. NB: we are using dataset "directory partitioning" to automatically get the `model_id`
-        # column from directory names
+        # column from directory names. regarding performance on S3-based datasets, we set `exclude_invalid_files=False`,
+        # which speeds up pyarrow's dataset processing, but opens the door to errors: "unsupported files may be present
+        # in the Dataset (resulting in an error at scan time)". we prevent this from happening by manually constructing
+        # and passing `ignore_prefixes` based on file extensions. this method accepts `ignore_files` to allow custom
+        # prefixes to ignore. it defaults to common ones for hubs
 
         # NB: we force file_formats to .parquet if not a LocalFileSystem (e.g., an S3FileSystem). otherwise we use the
         # list from self.admin['file_format']
         file_formats = ['parquet'] if not isinstance(self._filesystem, fs.LocalFileSystem) \
             else self.admin['file_format']
-        schema = create_hub_schema(self.tasks)
-        datasets = [ds.dataset(self.model_output_dir, filesystem=self._filesystem, format=file_format,
-                               partitioning=['model_id'],  # NB: hard-coded partitioning!
-                               exclude_invalid_files=True, schema=schema)
-                    for file_format in file_formats]
+        model_out_files = self._list_model_out_files()  # model_output_dir, type='file'
+        datasets = []
+        file_format_to_ignore_files: dict[str, list[fs.FileInfo]] = {}  # for warning
+        for file_format in file_formats:
+            _ignore_files = self._list_invalid_format_files(model_out_files, file_format, ignore_files)
+            file_format_to_ignore_files[file_format] = _ignore_files
+            dataset = ds.dataset(self.model_output_dir, filesystem=self._filesystem, format=file_format,
+                                 schema=self.schema, partitioning=['model_id'],  # NB: hard-coded partitioning!
+                                 exclude_invalid_files=False,
+                                 ignore_prefixes=[file_info.base_name for file_info in _ignore_files])
+            datasets.append(dataset)
         datasets = [dataset for dataset in datasets if len(dataset.files) != 0]
+        self._warn_unopened_files(model_out_files, ignore_files, file_format_to_ignore_files)
```

Comment:
Will this issue a warning everytime if a `.DS_Store` file is ignored? If so, I don't think this is something the user will be interested in.